pyproject.toml

```diff
@@ -27,14 +27,74 @@ repository = "https://github.com/xapi-project/xen-api"
 [tool.black]
 line-length = 88
 
+
+# -----------------------------------------------------------------------------
+# Coverage.py - https://coverage.readthedocs.io/en/coverage-5.5/config.html
+# -----------------------------------------------------------------------------
+
+[tool.coverage.report]
+# Here, developers can configure which lines do not need to be covered by tests:
+exclude_lines = [
+    "pragma: no cover",  # standard pragma for not covering a line or block
+    "if TYPE_CHECKING:", # imports for type checking only
+    "pass",
+    # Other specific lines that do not need to be covered, comment in which file:
+]
+# Here, developers can configure the minimum coverage required for the checks to pass:
+fail_under = 60
```

Comment:
Seems this is the reason why the Python coverage failed.
And there is another Python coverage configuration in .codecov.yml.

Reply:
Ah, I see. This would be useful when run locally, but it's a hindrance when run on CI, I've removed it